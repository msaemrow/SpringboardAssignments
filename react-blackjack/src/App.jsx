import { useState } from 'react'
import Game from "./components/Game"
import './App.css'

function App() {
  return (
    <div>
      <Game username="cardShark"/>
    </div>
  )
}

export default App;
