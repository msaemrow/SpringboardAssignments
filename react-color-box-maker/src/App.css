#root {
  margin: 10px auto;
  text-align: center;
  display: flex;
  flex-direction: column;
}
