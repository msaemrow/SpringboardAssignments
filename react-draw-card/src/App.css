body{
  background-color: green;
}

#root {
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.App-title{
  color: white;
}

