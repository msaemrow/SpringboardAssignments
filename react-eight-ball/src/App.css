body{
  background-color: rgb(188, 75, 237);
}

#root {
  max-width: 1280px;
  margin: 0 auto;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.App-header{
  margin:0 auto;
  font-size: 1.5rem;
}

.App-instructions{
  margin:5px auto;
  font-size: 0.9rem;
}
