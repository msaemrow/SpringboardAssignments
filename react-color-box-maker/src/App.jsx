import { useState } from 'react'
import './App.css'
import BoxList from "./components/BoxList"

function App() {
  const [count, setCount] = useState(0)

  return (
    <div>
      <h1>Create a color box!</h1>
      <BoxList />
    </div>

  )
}

export default App
