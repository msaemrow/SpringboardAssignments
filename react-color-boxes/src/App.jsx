import { React, useState } from 'react'
import BoxContainer from './components/BoxContainer';
import Box from "./components/Box";
import "./App.css"

function App() {
  return (
    <div>
      <BoxContainer />
    </div>
  )
}

export default App;
