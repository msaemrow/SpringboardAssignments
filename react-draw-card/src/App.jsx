import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import Deck from "./components/Deck"
import Deckv2 from "./components/Deckv2"

// Deck is a manual card draw component
// Deckv2 is an automatic card draw component

function App() {
  return (
    <>
      <h1 className='App-title'>AUTOMATIC CARD DRAW</h1>
      <Deckv2 />
    </>
  )
}

export default App
