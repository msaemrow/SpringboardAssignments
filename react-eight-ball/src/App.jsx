import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import answers from "./answer.json"
import EightBall from './components/EightBall'


function App() {
  return (
    <>
      <h1 className='App-header'>Magic 8 Ball!</h1>
      <p className='App-instructions'>Click the edge of the ball to reveal your answer</p>
      <EightBall answers={answers} />
    
    </>
  )
}

export default App
