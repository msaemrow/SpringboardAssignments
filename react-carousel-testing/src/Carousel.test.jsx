import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import App from "./App"
import TEST_IMAGES from "./_testCommon.js";
import { expect, it } from "vitest";

it("loads App properly", () => {
  render(<App />)
})

it("loads Carousel without crashing", () => {
  render(<Carousel
    photos={TEST_IMAGES}
    title="images for testing"
  />);

})

it('matches snapshot', () => {
  const { asFragment } = render(
    <Carousel
    photos={TEST_IMAGES}
    title="images for testing"
  />);
  expect(asFragment()).toMatchSnapshot();
})

it("works when you click on the right arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".arrow-right");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});


it("works when you click on the left arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // move forward in the carousel
  const rightArrow = container.querySelector(".arrow-right");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();

  const leftArrow = container.querySelector(".arrow-left");
fireEvent.click(leftArrow);
expect(
  container.querySelector('img[alt="testing image 1"]')
).toBeInTheDocument();
});


it("removes left arrow if on first image", () => {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector(".arrow-left")
  ).not.toBeInTheDocument();
})

it("removes left right arrow if on last image", () => {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  const rightArrow = container.querySelector(".arrow-right");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);
  expect(
    container.querySelector('img[alt="testing image 3"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector(".arrow-right")
  ).not.toBeInTheDocument();
})