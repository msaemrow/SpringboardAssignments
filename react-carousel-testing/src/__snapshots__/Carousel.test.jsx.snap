// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      images for testing
    </h1>
    <div
      class="Carousel-main"
    >
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 1 of 3.
        </small>
      </div>
      <button
        class="btn btn-primary arrow-right"
      >
        <svg
          class="Carousel-right-arrow bi bi-arrow-right"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
